datamover/pkg/drivers/https/prepare.go

```diff
@@ -9,143 +9,44 @@ import (
 	"time"
 
 	flowtype "github.com/opensds/multi-cloud/dataflow/pkg/model"
-	"github.com/opensds/multi-cloud/datamover/pkg/db"
 	. "github.com/opensds/multi-cloud/datamover/pkg/utils"
 	pb "github.com/opensds/multi-cloud/datamover/proto"
 	osdss3 "github.com/opensds/multi-cloud/s3/proto"
 	log "github.com/sirupsen/logrus"
 )
 
-func getCtxTimeout() time.Duration {
-	// 1 day as default
-	tmout := 86400 * time.Second
+func GetCtxTimeout(key string, min, max, def int64) time.Duration {
+	// default value
+	tmout := time.Duration(def) * time.Second
 
-	tmoutCfg, err := strconv.ParseInt(os.Getenv("JOB_MAX_RUN_TIME"), 10, 64)
```

Comment:
this is good, can be made generic like this ? 
`
func GetEnvVar(key string, min, max, def int64, varName string) {}
`
and use like this
`
getEnvVar(key, 10, 200, "JOB_MAX_RUN_TIME")
`

Reply:
Hi @rhsakarpos isn't key the same as varName?

Reply:
oh...yes it is. so inside this function, there is no specific key like "JOB_MAX_RUN_TIME"

Reply:
updated my initial comment to reflect this

Reply:
Yes, it is generic as what you said.